drivers/spi/Kconfig.nrfx

```diff
@@ -6,6 +6,7 @@
 menuconfig SPI_NRFX
 	bool "nRF SPI nrfx drivers"
 	depends on SOC_FAMILY_NRF
+	select HAS_DTS_SPI
```

Comment:
This option does not, really, belong here. HAS_DTS_SPI signifies that the target platform supports DTS for SPI configuration. It should be somehow selected by the Board or SOC definition.

I saw that the same mistake - IMHO - is done by similar platforms.

Reply:
Kconfig and DTS are currently connected in Zephyr. Driver's Kconfig needs to be aware if the DTS supports configuration of some peripheral because otherwise it can provide it's own defaults etc.

If you wish to discuss/modify this approach, let's don't do it in this PR.

Reply:
>Kconfig and DTS are currently connected in Zephyr. Driver's Kconfig needs to be aware if the DTS supports configuration of some peripheral because otherwise it can provide it's own defaults etc.

I didn't disagree on that; i stressed that DTS support should be provided in the hardware definition instead of the driver selection.

Reply:
I've acknowledged that this convention spreads out to other SOC/Board vendors

Reply:
Discussed offline with @anangl. Looks like this convention was introduced as the way to make contributors provide DTS configuration needed by drivers/peripherals, in their platforms (@galak , pls, correct if I am wrong).